{"name": "test_questions_and_answer[4-\\u041f\\u043e\\u043a\\u0430 \\u0447\\u0442\\u043e \\u043d\\u0435\\u0442! \\u041d\\u043e \\u0435\\u0441\\u043b\\u0438 \\u0447\\u0442\\u043e-\\u0442\\u043e \\u0441\\u0440\\u043e\\u0447\\u043d\\u043e\\u0435 \\u2014 \\u0432\\u0441\\u0435\\u0433\\u0434\\u0430 \\u043c\\u043e\\u0436\\u043d\\u043e \\u043f\\u043e\\u0437\\u0432\\u043e\\u043d\\u0438\\u0442\\u044c \\u0432 \\u043f\\u043e\\u0434\\u0434\\u0435\\u0440\\u0436\\u043a\\u0443 \\u043f\\u043e \\u043a\\u0440\\u0430\\u0441\\u0438\\u0432\\u043e\\u043c\\u0443 \\u043d\\u043e\\u043c\\u0435\\u0440\\u0443 1010.]", "status": "broken", "statusDetails": {"message": "TypeError: unsupported callable", "trace": "func = <staticmethod(<function BasePage.extraction_locator at 0x000001A668720E00>)>\n\n    def getfullargspec(func):\n        \"\"\"Get the names and default values of a callable object's parameters.\n    \n        A tuple of seven things is returned:\n        (args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, annotations).\n        'args' is a list of the parameter names.\n        'varargs' and 'varkw' are the names of the * and ** parameters or None.\n        'defaults' is an n-tuple of the default values of the last n parameters.\n        'kwonlyargs' is a list of keyword-only parameter names.\n        'kwonlydefaults' is a dictionary mapping names from kwonlyargs to defaults.\n        'annotations' is a dictionary mapping parameter names to annotations.\n    \n        Notable differences from inspect.signature():\n          - the \"self\" parameter is always reported, even for bound methods\n          - wrapper chains defined by __wrapped__ *not* unwrapped automatically\n        \"\"\"\n        try:\n            # Re: `skip_bound_arg=False`\n            #\n            # There is a notable difference in behaviour between getfullargspec\n            # and Signature: the former always returns 'self' parameter for bound\n            # methods, whereas the Signature always shows the actual calling\n            # signature of the passed object.\n            #\n            # To simulate this behaviour, we \"unbind\" bound methods, to trick\n            # inspect.signature to always return their first parameter (\"self\",\n            # usually)\n    \n            # Re: `follow_wrapper_chains=False`\n            #\n            # getfullargspec() historically ignored __wrapped__ attributes,\n            # so we ensure that remains the case in 3.3+\n    \n>           sig = _signature_from_callable(func,\n                                           follow_wrapper_chains=False,\n                                           skip_bound_arg=False,\n                                           sigcls=Signature,\n                                           eval_str=False)\n\nC:\\python\\Lib\\inspect.py:1385: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\nC:\\python\\Lib\\inspect.py:2592: in _signature_from_callable\n    return _signature_from_builtin(sigcls, obj,\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\ncls = <class 'inspect.Signature'>, func = <staticmethod(<function BasePage.extraction_locator at 0x000001A668720E00>)>, skip_bound_arg = False\n\n    def _signature_from_builtin(cls, func, skip_bound_arg=True):\n        \"\"\"Private helper function to get signature for\n        builtin callables.\n        \"\"\"\n    \n        if not _signature_is_builtin(func):\n            raise TypeError(\"{!r} is not a Python builtin \"\n                            \"function\".format(func))\n    \n        s = getattr(func, \"__text_signature__\", None)\n        if not s:\n>           raise ValueError(\"no signature found for builtin {!r}\".format(func))\nE           ValueError: no signature found for builtin <staticmethod(<function BasePage.extraction_locator at 0x000001A668720E00>)>\n\nC:\\python\\Lib\\inspect.py:2382: ValueError\n\nThe above exception was the direct cause of the following exception:\n\nself = <tests.test_main_page.TestMainPage object at 0x000001A6687247D0>, driver = <selenium.webdriver.firefox.webdriver.WebDriver (session=\"bfb0b037-2726-405f-b592-4c35ebcbb4b1\")>\nq_num = 4, expected_result = 'Пока что нет! Но если что-то срочное — всегда можно позвонить в поддержку по красивому номеру 1010.'\n\n    @allure.step('Тестируем вопрос-ответ')\n    @pytest.mark.parametrize(\"q_num, expected_result\",\n                             [(0, ANSWER_Q_0),\n                              (1, ANSWER_Q_1),\n                              (2, ANSWER_Q_2),\n                              (3, ANSWER_Q_3),\n                              (4, ANSWER_Q_4),\n                              (5, ANSWER_Q_5),\n                              (6, ANSWER_Q_6),\n                              (7, ANSWER_Q_7)])\n    def test_questions_and_answer(self, driver, q_num, expected_result):\n        main_page = MainPage(driver)\n        main_page.get_url(MAIN_URL)\n        main_page.click_on_cookie()\n>       result = main_page.click_to_question_get_answer_text(q_num)\n\ntests\\test_main_page.py:23: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npages\\main_page.py:24: in click_to_question_get_answer_text\n    self.click_on_question(num)\npages\\main_page.py:14: in click_on_question\n    question_loc = self.extraction_locator(MainPageLocators.QUESTION, num)\nvenv\\Lib\\site-packages\\allure_commons\\utils.py:231: in func_parameters\n    arg_spec = inspect.getfullargspec(func)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nfunc = <staticmethod(<function BasePage.extraction_locator at 0x000001A668720E00>)>\n\n    def getfullargspec(func):\n        \"\"\"Get the names and default values of a callable object's parameters.\n    \n        A tuple of seven things is returned:\n        (args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, annotations).\n        'args' is a list of the parameter names.\n        'varargs' and 'varkw' are the names of the * and ** parameters or None.\n        'defaults' is an n-tuple of the default values of the last n parameters.\n        'kwonlyargs' is a list of keyword-only parameter names.\n        'kwonlydefaults' is a dictionary mapping names from kwonlyargs to defaults.\n        'annotations' is a dictionary mapping parameter names to annotations.\n    \n        Notable differences from inspect.signature():\n          - the \"self\" parameter is always reported, even for bound methods\n          - wrapper chains defined by __wrapped__ *not* unwrapped automatically\n        \"\"\"\n        try:\n            # Re: `skip_bound_arg=False`\n            #\n            # There is a notable difference in behaviour between getfullargspec\n            # and Signature: the former always returns 'self' parameter for bound\n            # methods, whereas the Signature always shows the actual calling\n            # signature of the passed object.\n            #\n            # To simulate this behaviour, we \"unbind\" bound methods, to trick\n            # inspect.signature to always return their first parameter (\"self\",\n            # usually)\n    \n            # Re: `follow_wrapper_chains=False`\n            #\n            # getfullargspec() historically ignored __wrapped__ attributes,\n            # so we ensure that remains the case in 3.3+\n    \n            sig = _signature_from_callable(func,\n                                           follow_wrapper_chains=False,\n                                           skip_bound_arg=False,\n                                           sigcls=Signature,\n                                           eval_str=False)\n        except Exception as ex:\n            # Most of the times 'signature' will raise ValueError.\n            # But, it can also raise AttributeError, and, maybe something\n            # else. So to be fully backwards compatible, we catch all\n            # possible exceptions here, and reraise a TypeError.\n>           raise TypeError('unsupported callable') from ex\nE           TypeError: unsupported callable\n\nC:\\python\\Lib\\inspect.py:1395: TypeError"}, "steps": [{"name": "Тестируем вопрос-ответ", "status": "broken", "statusDetails": {"message": "TypeError: unsupported callable\n", "trace": "  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\tests\\test_main_page.py\", line 23, in test_questions_and_answer\n    result = main_page.click_to_question_get_answer_text(q_num)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\pages\\main_page.py\", line 24, in click_to_question_get_answer_text\n    self.click_on_question(num)\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\pages\\main_page.py\", line 14, in click_on_question\n    question_loc = self.extraction_locator(MainPageLocators.QUESTION, num)\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 189, in impl\n    params = func_parameters(func, *a, **kw)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\utils.py\", line 231, in func_parameters\n    arg_spec = inspect.getfullargspec(func)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\python\\Lib\\inspect.py\", line 1395, in getfullargspec\n    raise TypeError('unsupported callable') from ex\n"}, "steps": [{"name": "Подгружаем драйвер в конструктор", "status": "passed", "parameters": [{"name": "driver", "value": "<selenium.webdriver.firefox.webdriver.WebDriver (session=\"bfb0b037-2726-405f-b592-4c35ebcbb4b1\")>"}], "start": 1722356822484, "stop": 1722356822484}, {"name": "Загружаем страницу", "status": "passed", "parameters": [{"name": "url", "value": "'https://qa-scooter.praktikum-services.ru/'"}], "start": 1722356822484, "stop": 1722356824439}, {"name": "Соглашаемся на использование куки", "status": "passed", "steps": [{"name": "Кликаем на элемент", "status": "passed", "steps": [{"name": "Ищем элемент на странице", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', '//button[text() = \"да все привыкли\"]')"}], "start": 1722356824439, "stop": 1722356824477}], "parameters": [{"name": "locator", "value": "('xpath', '//button[text() = \"да все привыкли\"]')"}], "start": 1722356824439, "stop": 1722356824724}], "start": 1722356824439, "stop": 1722356824724}, {"name": "Кликаем на вопрос и получаем текст ответа", "status": "broken", "statusDetails": {"message": "TypeError: unsupported callable\n", "trace": "  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\pages\\main_page.py\", line 24, in click_to_question_get_answer_text\n    self.click_on_question(num)\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\pages\\main_page.py\", line 14, in click_on_question\n    question_loc = self.extraction_locator(MainPageLocators.QUESTION, num)\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 189, in impl\n    params = func_parameters(func, *a, **kw)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\utils.py\", line 231, in func_parameters\n    arg_spec = inspect.getfullargspec(func)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\python\\Lib\\inspect.py\", line 1395, in getfullargspec\n    raise TypeError('unsupported callable') from ex\n"}, "steps": [{"name": "Кликаем на вопрос", "status": "broken", "statusDetails": {"message": "TypeError: unsupported callable\n", "trace": "  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\pages\\main_page.py\", line 14, in click_on_question\n    question_loc = self.extraction_locator(MainPageLocators.QUESTION, num)\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 189, in impl\n    params = func_parameters(func, *a, **kw)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\user\\YandexProjects\\Sprint_6\\venv\\Lib\\site-packages\\allure_commons\\utils.py\", line 231, in func_parameters\n    arg_spec = inspect.getfullargspec(func)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\python\\Lib\\inspect.py\", line 1395, in getfullargspec\n    raise TypeError('unsupported callable') from ex\n"}, "parameters": [{"name": "num", "value": "4"}], "start": 1722356824724, "stop": 1722356824725}], "parameters": [{"name": "num", "value": "4"}], "start": 1722356824724, "stop": 1722356824726}], "parameters": [{"name": "driver", "value": "<selenium.webdriver.firefox.webdriver.WebDriver (session=\"bfb0b037-2726-405f-b592-4c35ebcbb4b1\")>"}, {"name": "q_num", "value": "4"}, {"name": "expected_result", "value": "'Пока что нет! Но если что-то срочное — всегда можно позвонить в поддержку по красивому номеру 1010.'"}], "start": 1722356822484, "stop": 1722356824728}], "parameters": [{"name": "q_num", "value": "4"}, {"name": "expected_result", "value": "'Пока что нет! Но если что-то срочное — всегда можно позвонить в поддержку по красивому номеру 1010.'"}], "start": 1722356822484, "stop": 1722356824730, "uuid": "5b1d2a67-c0de-429f-8813-a5b45fbc1d16", "historyId": "51fd8b047e6de430318ac0a5e6c3ebee", "testCaseId": "c8847cf02ab9a3414715e33c97d25583", "fullName": "tests.test_main_page.TestMainPage#test_questions_and_answer", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_main_page"}, {"name": "subSuite", "value": "TestMainPage"}, {"name": "host", "value": "WIN-O4R89KV0872"}, {"name": "thread", "value": "3352-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_main_page"}]}